src/core/lib/promise/mpsc.h

```diff
@@ -204,6 +204,27 @@ class MpscReceiver {
     };
   }
 
+  // Return a promise that will resolve to a vector containing all available
+  // items.
+  auto AllNext() {
+    return [this]() -> Poll<std::vector<T>> {
+      if (buffer_it_ == buffer_.end()) {
+        if (center_->PollReceiveBatch(buffer_)) {
+          buffer_it_ = buffer_.begin();
```

Comment:
I think this could be `return std::move(buffer_);` -- avoiding the loop & move below

Reply:
Fixed it now.